### PR TITLE
dev(debug): complete dev environment to debug/code easily

### PR DESCRIPTION
**Pull request by yarysp**
To help develop reNgine I've made 2 environments :

- docker-compose.dev.yml :
    This docker compose set up all the things to activate debug tools : **Django debug toolbar**, **remote debug** (disabled by default) via **VSCode/Python Debugger extensions**, more debug information for **Celery** in make logs
    It uses **Django development server**
    If you activate Remote debug, there are 2 behaviors, depending on whether you are debugging the **webUI** or the tasks
  - **webUI** : Attach the VSCode debugguer to the 5678 port, set a break point, browse on the **webUI**, it stops when the breakpoint is reached
  - **Celery** : It goes in monotask, only 1 concurrency thread.       
        And when you launch task from the webUI, it hangs on the first line of the InitiateScan or InitiateSubscan tasks and wait for the VSCode debugger to attach to 5679 port to step debug inside the task
        
image

- docker-compose.yml
    This one is the production environment, it uses **gunicorn** as web server instead of the **Django development server**

I've added the necessary command in MakeFile to easily startup dev or prod environment

To start dev
```sh
make dev_up
```

To start prod
```sh
make up
```

To restart in dev from prod

```sh
make down
make dev_up
```

To restart in prod from dev

```sh
make dev_down
make up
```

### CODE REVIEW

**Comment by psyray** [comment]
Tests done, all is working well

Django debug toolbar
image

File: Makefile [hunk 1]
```diff
@@ -58,3 +60,15 @@ prune:			## Remove containers and delete volume data.
...
+	${COMPOSE_PREFIX_CMD} docker-compose ${COMPOSE_DEV_ALL_FILES} build ${SERVICES_DEV}
+
+dev_up:				## Build and start all services.
+	${COMPOSE_PREFIX_CMD} docker-compose ${COMPOSE_DEV_ALL_FILES} up -d --build ${SERVICES_DEV}
+
+dev_down:			## Down all services.
+	${COMPOSE_PREFIX_CMD} docker-compose ${COMPOSE_DEV_ALL_FILES} down
+
+dev_logs:			## Tail all logs with -n 1000.
+	${COMPOSE_PREFIX_CMD} docker-compose $(COMPOSE_DEV_ALL_FILES) logs --follow --tail=1000 ${SERVICES_DEV}
```
Comment (AnonymousWP) [inline Makefile:74]: Please change everything from `docker-compose` to the 'new' syntax `docker compose`, see https://stackoverflow.com/a/66516826. I thought I changed it in some PR I sent to Yogesh's repo, but it's not here nor applied it seems.

File: README.md [hunk 2]
```diff
@@ -398,6 +398,39 @@ See the [Contributing Guide](.github/CONTRIBUTING.md) to get started.
...
+### Submitting issues
+
+You can submit issues related to this project, but you should do it in a way that helps developers to resolve it as quickly as possible.
+
+For that, you need to add as much valuable information as possible.
+
+You can have this valuable information by following these steps:
+
+- Go to the root of the git cloned project
+- Shutdown you current production instance by launching `make down`
```
Comment (AnonymousWP) [inline README.md:410]:
```suggestion
- Shutdown your current production instance by typing `make down`
```

File: README.md [hunk 3]
```diff
@@ -398,6 +398,39 @@ See the [Contributing Guide](.github/CONTRIBUTING.md) to get started.
...
+- Launch the dev environment by launching `make dev_up`
```
Comment (AnonymousWP) [inline README.md:411]:
```suggestion
- Launch the dev environment by typing `make dev_up`
```

File: README.md [hunk 4]
```diff
@@ -398,6 +398,39 @@ See the [Contributing Guide](.github/CONTRIBUTING.md) to get started.
...
+- Then you can start `make logs` and run into your issue, you should have more detailed log (stack trace ...)
```
Comment (AnonymousWP) [inline README.md:412]:
```suggestion
- Then you can start `make logs` and run into your issue, you should now have a more detailed log (stack trace ...)
```

**Review by AnonymousWP — CHANGES_REQUESTED** [review]
Awesome work! Just a few nitpicks.

File: Makefile (hunk 1 above)
Comment (AnonymousWP) [reply Makefile:74]: Found it: https://github.com/yogeshojha/rengine/pull/1161

Comment (AnonymousWP) [reply Makefile:74]: You can [cherry-pick](https://www.atlassian.com/git/tutorials/cherry-pick) that syntax modification commit into this branch. Then, we can merge this into master. 🙂 That way, you can modify that specific line of code, or you can backport it manually.

File: install.sh
Comment (AnonymousWP) [inline install.sh:1]: Line 29 (https://github.com/Security-Tools-Alliance/rengine-ng/pull/68/files#diff-043df5bdbf6639d7a77e1d44c5226fd7371e5259a1e4df3a0dd5d64c30dca44fR29) is causing an issue upon testing for me. If you remove the backticks then it'll be fine, I just tested that.

**Review by AnonymousWP — CHANGES_REQUESTED** [review]

**Comment by AnonymousWP** [comment]
If there's an issue of this feature, then please link it.

File: Makefile (hunk 1 above)
Comment (psyray) [reply Makefile]: Done

**Comment by psyray** [comment]
PR rebased on latest master version
Ready to merge

**Review by AnonymousWP — CHANGES_REQUESTED** [review]
Two more backticks at line 20. 🙂 https://github.com/yarysp/rengine-ng/blob/debug-environment/install.sh#L20

**Pull request merged**